src/main/java/ca/corefacility/bioinformatics/irida/ria/web/linelist/LineListController.java

```diff
@@ -127,18 +123,67 @@ public String saveMetadataEntry(@RequestParam long sampleId, @RequestParam Strin
 		}
 	}
 
+	/**
+	 * Remove a {@link MetadataTemplateField} from all {@link Sample}s in a {@link Project}
+	 *
+	 * @param label     {@link String} the name of the {@link MetadataTemplateField}
+	 * @param projectId {@link Long} identifier for the {@link Project}
+	 * @param principal {@link Principal}
+	 * @param locale {@link Locale}
+	 * @return {@link UIRemoveMetadataFieldResponse} a description of what happened when the entries were removed
+	 */
+	@RequestMapping(value = "/entries", method = RequestMethod.DELETE)
+	@ResponseBody
+	public UIRemoveMetadataFieldResponse removeDataFromMetadataField(@RequestParam String label,
+			@RequestParam Long projectId, Principal principal, Locale locale) {
+		this.permissions.setPrincipalPermissions(principal);
+		Project project = projectService.read(projectId);
+		MetadataTemplateField field = metadataTemplateService.readMetadataFieldByLabel(label);
+		List<Join<Project, Sample>> projectSampleJoins = sampleService.getSamplesForProject(project);
+		int lockedCount = 0;
+		for (Join<Project, Sample> join : projectSampleJoins) {
```

Comment:
Can all this functionality be built into a service method marked as `@Transactional` please? Currently if there was an issue with one sample it wouldn't properly handle a failure.

Reply:
Updated in 4dd14c52a794a03a108933e7d7b5ff0cd3d90aa5